resdk/analysis/prepare_geo.py

```diff
@@ -0,0 +1,160 @@
+"""Prepare GEO."""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import time
+import traceback
+
+from resdk.resources.utils import (
+    get_resource_collection, get_samples, get_resolwe
+)
+
+__all__ = ('prepare_geo_chipseq', 'prepare_geo_rnaseq', 'prepare_geo')
+
+
+def is_background(sample):
+    """Return ``True`` if given sample is background and ``False`` otherwise."""
+    background_relations = sample.resolwe.relation.filter(
+        type='compare',
+        label='background',
+        entity=sample.id,
+        position='background'
+    )
+    return len(background_relations) > 0
+
+
+def get_name_collection(collection_ids, res):
+    """Get the ``name`` input for ``prepare_geo`` processes."""
+    collection = None
+    if len(collection_ids) == 1:
+        collection_id = next(iter(collection_ids))
```

Comment:
You have already confirmed that `collection_ids` contains only one element: why not just use `collection_id = collection_ids[0]`? Also, checking the existance of `collection_id` (next line) does not make sense: list contains one element and you have just picked it up...

Reply:
See comment above